manager/manager.go

```diff
@@ -128,8 +128,15 @@ type Config struct {
 	// FIPS setting.
 	FIPS bool
 
-	// NetworkConfig stores network related config for the cluster
-	NetworkConfig *cnmallocator.NetworkConfig
+	// DefaultAddrPool specifies default subnet pool for global scope networks
+	DefaultAddrPool []string
```

Comment:
Previously, a pointer was used; are there situations where these could be nil (and for a reason?)

Actually wondering; should we just copy the `NetworkConfig` type here, and use that instead of defining these fields both here, and in node/node.go ?